wdl2cwl/tests/cwl_files/deepvariant.cwl

```diff
@@ -103,12 +96,12 @@ requirements:
             --reads $(inputs.inputBam.path) \
             --model_type $(inputs.modelType) \
             --output_vcf $(inputs.outputVcf) \
-            --output_gvcf $(inputs.outputGVcf) \
-            --customized_model $(inputs.customizedModel) \
-            --num_shards $(inputs.numShards) \
-            --regions $(inputs.regions) \
-            --sample_name $(inputs.sampleName) \
-            --postprocess_variants_extra_args $(inputs.postprocessVariantsExtraArgs) \
+            --output_gvcf $(inputs["outputGVcf"] === null ? "" : inputs["outputGVcf"]) \
```

Comment:
The original WDL expression is `~{"--output_gvcf " + outputGVcf} \` so the CWL version should be
```suggestion
            $(inputs["outputGVcf"] === null ? "" : "--output_gvcf " + inputs["outputGVcf"]) \
```

This is the tricky part.

If a WDL expression has even a single reference to an optional input (even if there are other, non-optional references) and that input is missing (`=== null`) then the entire WDL expression should result in an empty string.

Reply:
:/ That requires some work. What will happen if there's two variables inside the expression ```~{}``` where only one is optional and the other is a required variable. Should the entire expression still evaluate to `null`?

Reply:
> What will happen if there's two variables inside the expression ~{} where only one is optional and the other is a required variable. Should the entire expression still evaluate to null?

Yes. Tricky!